### PR TITLE
Add invert PDF in dark mode only.

### PR DESCRIPTION
**Pull request by aterenin**
This creates an option which applies "invert PDF" when the operating system's dark mode setting is enabled. 

This is detected via `window.matchMedia(`(prefers-color-scheme: dark)`).matches`, which is the same mechanism VS Code internally uses for the `window.autoDetectColorScheme` option which auto-switches between light and dark themes.

### CODE REVIEW

**Comment by tamuratak** [comment]
With this PR, the PDF Viewer does not get invert-mode when the theme of VS Code is dark and the mode of OS is not dark. It would be a good idea to detect the darkness of VS Code, not OS. 

See
- https://github.com/James-Yu/LaTeX-Workshop/blob/6bb95c2058bb2e5aaefccc37df04b174bab89d95/src/utils/theme.ts#L50 
- https://code.visualstudio.com/api/references/vscode-api#window.activeColorTheme

**Comment by aterenin** [comment]
Theme is now detected via `getCurrentThemeLightness()`. Let me know if this is OK.

**Comment by tamuratak** [comment]
No. `viewer/latexworkshop.ts` is executed in the browser context. You can not import `src/utils/theme`, which uses Node API.

**Comment by aterenin** [comment]
D'oh. So much for quickly trying to hack things together.

The conditional for `darkModeOnly` is now evaluated in the handler, which, depending on `getCurrentThemeLightness` either sends the user's invert value or zero.

**Comment by aterenin** [comment]
@tamuratak Any further comments? Is there still an issue with the current code?

**Comment by tamuratak** [comment]
Sorry for the delay in replying.

I was thinking of making the option, `"latex-workshop.view.pdf.invertMode.darkModeOnly"`, more intuitive. For example, make a new option `"latex-workshop.view.pdf.invertMode.enabled"`, which can be "auto", "always", "compat", and "never". 

`"auto"` is the same as your `darkModeOnly`. `"always"` enables the invert mode even when `invert === 0`. `"compat"` behaves in the same way as now, enabled only when `invert > 0`.

No conclusions have been reached yet. What do you think?

**Comment by aterenin** [comment]
That makes sense, and sounds more intuitive. Should I implement it?

**Comment by tamuratak** [comment]
I would appreciate you if you would implement it.

**Comment by aterenin** [comment]
> I would appreciate you if you would implement it.

Implemented! Let me know how this looks now.

File: package.json [hunk 1]
```diff
@@ -1272,7 +1272,24 @@
...
-        "latex-workshop.view.pdf.invert": {
+        "latex-workshop.view.pdf.invertMode.enabled": {
+          "type": "string",
+          "enum": [
+            "auto",
+            "always",
+            "compat",
+            "never"
+          ],
+          "default": "never",
```
Comment (tamuratak) [inline package.json:1283]: The default value should be "compat"

File: src/components/viewer.ts [hunk 2]
```diff
@@ -407,6 +408,10 @@ export class Viewer {
                         continue
                     }
                     const configuration = vscode.workspace.getConfiguration('latex-workshop')
+                    const invertType = configuration.get('view.pdf.invertMode.enabled') as string
+                    const invertEnabled = (invertType == "auto" && (getCurrentThemeLightness() == 'dark')) ||
+                        invertType == "always" ||
+                        (invertType == "compat" && ((configuration.get('view.pdf.invert') as number) > 0))
```
Comment (tamuratak) [inline src/components/viewer.ts:414]: Run `npm run lint:fix`

File: package.json (hunk 1 above)
Comment (aterenin) [reply package.json]: Fixed!

File: src/components/viewer.ts (hunk 2 above)
Comment (aterenin) [reply src/components/viewer.ts]: Done!

**Comment by tamuratak** [comment]
Thank you, again.

**Pull request merged**